src/Core/Exception/Exception.php

```diff
@@ -51,7 +51,7 @@ class Exception extends RuntimeException
      *
      * @var int
      */
-    protected $_defaultCode = 500;
+    protected $_defaultCode = 0;
```

Comment:
Should we use a non-zero value? `Cake\Error\ConsoleErrorHandler::handleException()` uses the exception code as an exit code which should be non-zero.

Reply:
I was considering making it `1` but started with 0 since that's the PHP default.

Does `handleException()` expect every exception error code to be > 0? There were a couple exceptions that already defaulted to 0.

Reply:
I wonder if that should really treat exceptions as exit codes. Maybe just returning 1 by default would be better even if we changed the exception default.

Different exceptions don't really mean the same exit code.

Reply:
Treating exception codes as exit codes is as wrong as treating all exception codes as http status codes :) 

So IMO we should keep exception codes as zero by default and make ConsoleErrorHandler return exit code 1.

Reply:
Ok. That is a behavior change we'll need to document though. Users could be using the exception code propagation to indicate to other tools the error state of a command/shell should it fail.

Reply:
We could create a base command center exception like http exception where the error code is treated as the exit code.

Reply:
What about using the existing Cake\Console\Exception\ConsoleException? We have a few subclasses of that including `StopException` which would ideally retain custom exit codes.

Reply:
Updated ConsoleErrorHandler to use error code from ConsoleException and updated unit tests to cover exit codes.